modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Meta.java

```diff
@@ -136,52 +141,38 @@ public void execute() {
      */
     private static Converter<SupportingFile, File> processFiles(final File targetDir,
             final Map<String, Object> data) {
-        return new Converter<SupportingFile, File>() {
-            private DefaultGenerator generator = new DefaultGenerator();
-
-            @Override
-            public File convert(SupportingFile support) {
-                try {
-                    File destinationFolder =
-                            new File(new File(targetDir.getAbsolutePath()), support.folder);
-                    File outputFile = new File(destinationFolder, support.destinationFilename);
-
-                    String template =
-                            generator.readTemplate(new File(TEMPLATE_DIR_CLASSPATH,
-                                    support.templateFile).getPath());
-                    String formatted = template;
-
-                    if (support.templateFile.endsWith(MUSTACHE_EXTENSION)) {
-                        LOGGER.info("writing file to {}", outputFile.getAbsolutePath());
-                        formatted =
-                                Mustache.compiler().withLoader(loader(generator)).defaultValue("")
-                                        .compile(template).execute(data);
-                    } else {
-                        LOGGER.info("copying file to {}", outputFile.getAbsolutePath());
-                    }
-
-                    FileUtils.writeStringToFile(outputFile, formatted, StandardCharsets.UTF_8);
-                    return outputFile;
-
-                } catch (IOException e) {
-                    throw new RuntimeException("Can't generate project", e);
+        return support -> {
+            try {
+                File destinationFolder =
+                        new File(new File(targetDir.getAbsolutePath()), support.folder);
+                File outputFile = new File(destinationFolder, support.destinationFilename);
+
+                TemplateManager templateProcessor = new TemplateManager(
+                        new TemplateManagerOptions(false, false),
+                        new MustacheEngineAdapter(),
+                        new TemplatePathLocator[]{ new CommonTemplateContentLocator("codegen") }
+                );
+
+                String template = templateProcessor.readTemplate(new File(TEMPLATE_DIR_CLASSPATH, support.templateFile).getPath());
+
+                String formatted = template;
+
+                Mustache.TemplateLoader loader = name -> templateProcessor.getTemplateReader(name.concat(MUSTACHE_EXTENSION));
```

Comment:
Notice regarding the design… template processor no longer requires that we pass that logic through a `DefaultGenerator` instance, which holds logic for OpenAPI input generation to templated output.

We can now generate without tight coupling to OpenAPI documents. This is work directed toward #841 and somewhat related to #843.